components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/ciba/CibaAuthRequestValidator.java

```diff
@@ -0,0 +1,773 @@
+
+            // Getting expected audience dynamically from configs.
+            String expectedAudience = OAuth2Util.getIdTokenIssuer(domain);
+
+            // Validation for aud-audience to meet mandated value.
+            if (!aud.contains(expectedAudience)) {
+                // The audience value does not suit mandated value.
+                if (log.isDebugEnabled()) {
+                    log.debug("Invalid CIBA Authentication Request made by client with clientID : " + clientId +
+                            ".Invalid value for (aud).");
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.INVALID_REQUEST,
+                        "parameter (aud) does not meet the mandated value.");
+            }
+        } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
+            throw new CibaAuthFailureException(OAuth2ErrorCodes.SERVER_ERROR, "Error in validating for (aud).", e);
+        }
+    }
+
+    /**
+     * Checks whether the client is valid.
+     *
+     * @param request CIBA Authentication request.
+     * @throws CibaAuthFailureException CIBA Authentication Failed Exception.
+     */
+    public void validateClient(String request) throws CibaAuthFailureException {
+
+        try {
+            SignedJWT signedJWT = SignedJWT.parse(request);
+            JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+            String clientId = claimsSet.getIssuer();
+
+            // Validate whether the claim for the issuer exists.
+            if (clientId == null) {
+                // ClientID does not exist.
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Missing issuer of the JWT of the request : " + request);
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT,
+                        "missing mandated parameter (iss).");
+            }
+
+            // Validate whether the claim for the issuer is a valid value.
+            if (StringUtils.isBlank(claimsSet.getIssuer())) {
+                // (iss) is a blank value.
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Missing issuer of the JWT of the request : " + request);
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT, "invalid (iss) parameter.");
+            }
+
+            // Validation for existence of clientApp  in the store.
+            OAuthAppDO appDO = OAuth2Util.getAppInformationByClientId(clientId);
+            String clientSecret = appDO.getOauthConsumerSecret();
+            if (StringUtils.isBlank(clientSecret)) {
+
+                if (log.isDebugEnabled()) {
+                    log.debug("The request : " + request + " doesn't have a proper clientID.");
+                }
+                throw new CibaAuthFailureException(OAuth2ErrorCodes.UNAUTHORIZED_CLIENT, "unknown (iss) client.");
+            }
```

Comment:
Why do we need to check this?

Reply:
Isn't appDO != null sufficient to prove the validity of the clientId

Reply:
An additional check we could do would be to check if CIBA grant type is enabled for the client app here.

Reply:
Ack